mlir/lib/Dialect/MIOpen/Transforms/AffixTuningParameters.cpp

```diff
@@ -37,6 +37,13 @@ struct AffixTuningParameters : public MIOpenOpsAffixTuningParametersPassBase<Aff
 
   // Actual implementation.
   template <typename T> void affixTuningParametersImpl(T &op);
+
+  void affixBackwardWeightUtilityKernels(miopen::Conv2DBwdWeightOp &op);
+
+  template <typename T>
+  std::tuple<int64_t, int64_t, int64_t, int64_t, int64_t, int64_t, int64_t,
+             int64_t, int64_t>
+  fetchDimensions(T &op);
```

Comment:
Any particular reason this is a template? You could either take an `Operation *` or, in things we probably should've done a while ago, define a trait that all the convolution ops share.

Reply:
Same reason `affixTuningParametersImpl` is a template. Different ops will invoke `fetchDimensions`. And we are explicitly walking on those different ops in `AffixTuningParameters::runOnOperation()`.

Using a trait is an interesting option which we can explore in https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/439

Reply:
Fair enough, I haven't poked this stuff in a detail, and so on.